pages/common/xargs.md

```diff
@@ -1,19 +1,20 @@
 # xargs
 
-> Execute a command with piped arguments.
+> Execute a command with piped arguments coming from another command, a file, etc.
+> The input is treated as a single block of text and split into separate arguments on spaces, tabs, newlines and end-of-file.
 
-- Main use:
+- Main usage pattern:
 
-`{{arguments}} | xargs {{command}}`
+`{{arguments_source}} | xargs {{command}}`
 
-- Specific example: delete all files that start with 'M':
+- Delete all files with a `.backup` extension:
 
-`find . -name 'M*' | xargs rm`
+`{{find . -name '*.backup'}} | xargs {{rm -v}}`
 
-- Handle whitespace in arguments:
+- Use the NUL (`\0`) character as separator (useful to split the input on newlines but not on spaces):
```

Comment:
Frankly, I am okay with both. @waldyrious' style is a bit more verbose, @ostera's version is a bit succinct. 

I think what would really help (IMO) is to explain in what scenario this option shines. And that is when the input contains spaces. Its a bit hard to grasp it without a concrete example. This would be my version -

```
Use the NUL (`\0`) character as separator (useful if a filename contains a space):
find . -type f | tr '\n' '\0' | xargs -0 {{command}}
```

I think this clearly explains "why" the `-0` flag is helpful. Just my 2 cents.